sink_snippets/src/g3sinks/ColorCoutSink.h

```diff
@@ -3,12 +3,28 @@
 #include <iostream>
 #include <g3log/logmessage.hpp>
 
-// NOTE: This works only on Linux/OSX
-// TODO KjellKod: For Windows terminals you can tweak this easily: https://stackoverflow.com/a/4053879/1066879
+#ifdef _WIN32
+
+#ifndef WIN32_LEAN_AND_MEAN
+    #define WIN32_LEAN_AND_MEAN 1
+#endif
+
+#include <Windows.h>
+#endif
+
 struct ColorCoutSink {
+#ifdef _WIN32
```

Comment:
I could make two different header files if you want to, originally it was supposed to have the ifdefs because I also added the filter code inside the ColorCoutSink (much like LogRotateWithFilter), hence why it was easier for me to have the ifdef.

Reply:
sure, if you don't mind. Either two header files or one file but keeping all the content separate. 
it's a headache to have to read `#ifdef`'s inside functions